Detectors/TRD/simulation/include/TRDSimulation/Detector.h

```diff
@@ -24,11 +24,55 @@ namespace o2
 {
 namespace trd
 {
-class HitType : public o2::BasicXYZEHit<float>
+class HitType : public o2::BaseHit
 {
+  Point3D<float> mPos; // cartesian position of Hit
+  float mTime;         // time of flight
+  int mCharge;         // energy loss
```

Comment:
the only difference seems to be to save "int" and not "float". Does this really necessitate a new type? Can you not save your int charges as float (energies)?
In order to at least reduce some of the code duplication you could also do a wrapper:

```
class HitType : protected BasicXYZEHit<float,float> {
public:
  HitType(float x, ...); 
 // list all interfaces that you want to use from Basic 
 using GetX;
  using GetY;
 // a wrapper interface
  int getCharge() const { return getEnergy(); }
}
```
I would also welcome a PR containing a more flexible BasicHit.

Reply:
Since we are working on this for a separate PR with a more flexible BasicHit type I will just close this PR.